src/workerd/jsg/util.h

```diff
@@ -394,4 +394,18 @@ inline bool isFinite(double value) {
   return !(kj::isNaN(value) || value == kj::inf() || value == -kj::inf());
 }
 
+// ======================================================================================
+
+class Lock;
+
+v8::MaybeLocal<v8::String> newExternalOneByteString(Lock& js, kj::ArrayPtr<const char> buf);
```

Comment:
This function will interpret `buf` as LATIN-1 rather than UTF-8, which is inconsistent with how KJ normally represents text. In particular, worker source code is definitely UTF-8, so passing it to this method would corrupt the code, unless it is limited to the ASCII subset. Many scripts probably are ASCII, so we could use this as an optimization if we've first determined that the script is ASCII, but if it's not ASCII then we either need to convert to UTF-16 and use the 2-byte version, or just give up and let V8 allocate the string as it normally would and do the conversion itself.

Reply:
Adding a comment for now.